lib/Stats.tsx

```diff
@@ -1,101 +1,236 @@
+  left: 24px;
 `
 
 interface StatsProps {
   readonly api: string
   readonly parameters: VapixParameters
   readonly videoProperties: VideoProperties
-  readonly host: string
   readonly open: boolean
   readonly refresh: number
 }
 
+interface Stat {
+  readonly name: string
+  readonly value: string | number
+  readonly unit?: string
+}
+
 export const Stats: React.FC<StatsProps> = ({
   api,
   parameters,
   videoProperties,
-  host,
   refresh,
 }) => {
-  const streamType = useMemo(() => {
-    if (api === 'jpg') {
-      return 'Still image'
-    }
+  const [stats, setStats] = useState<Array<Stat>>([])
+  const [showStats, setShowStats] = useState(true)
 
-    if (api === 'media') {
-      if (parameters['videocodec'] === 'h264') {
-        return 'H.264'
+  // Handles show/hide stats
+  const onToggleStats = useCallback(
+    (e) => {
+      e.preventDefault()
+      setShowStats((prevState) => !prevState)
+    },
+    [setShowStats],
+  )
+
+  // Updates stat values
+  const statsUpdateRef = useRef(0)
+  useEffect(() => {
+    const updateValues = () => {
+      let streamType = 'Unknown'
+      if (api === 'jpg') {
+        streamType = 'JPG'
+      } else if (api === 'media') {
+        if (parameters['videocodec'] === 'h264') {
+          streamType = 'RTSP (WebSocket)'
+        } else {
+          streamType = 'MJPG'
+        }
+      }
+      const { width, height, pipeline } = videoProperties
+      let statsData: Array<Stat> = [
+        {
+          name: 'Stream type',
+          value: streamType,
+        },
+        {
+          name: 'Resolution',
+          value: `${width}x${height}`,
+        },
+        {
+          name: 'Refreshed',
+          value: refresh,
+          unit: refresh > 1 ? 'times' : 'time',
+        },
+      ]
+      if (pipeline !== null && pipeline !== undefined) {
+        const tracks = pipeline.tracks?.map((track, index) =>
+          Object.assign({ index }, track),
+        )
+        const videoTrack = tracks?.find((track) => track.type === 'video')
+        if (videoTrack !== undefined) {
+          const { coding, profile, level } = videoTrack?.codec
+          const framerate = Number(
+            pipeline.framerate[videoTrack.index].toFixed(2),
+          )
+          const bitrate = Math.round(pipeline.bitrate[videoTrack.index] / 1000)
+
+          statsData = statsData.concat([
+            {
+              name: 'Encoding',
+              value: `${coding} ${profile} (${level})`,
+            },
+            {
+              name: 'Frame rate',
+              value: framerate,
+              unit: 'fps',
+            },
+            {
+              name: 'Bitrate',
+              value: bitrate,
+              unit: 'kbit/s',
+            },
+          ])
+        }
       }
-      return 'Motion JPEG'
+
+      setStats(statsData)
     }
 
-    return 'Unknown'
-  }, [api, parameters])
+    updateValues()
+
+    statsUpdateRef.current = window.setInterval(updateValues, 1000)
```

Comment:
Why is a ref needed here?

Reply:
for the `setInterval()`

Reply:
Could we use setInterval from https://github.com/AxisCommunications/react-hooks-shareable?

Reply:
Done